springApp/src/main/java/com/kuchumov/springApp/service/DataService.java

```diff
@@ -0,0 +1,316 @@
+package com.kuchumov.springApp.service;
+
+import com.kuchumov.springApp.DTO.*;
+import com.kuchumov.springApp.entity.DataModel;
+import com.kuchumov.springApp.exceptionHandler.customExceptions.FileIdNotFoundException;
+import com.kuchumov.springApp.exceptionHandler.customExceptions.ParsingDateException;
+import com.kuchumov.springApp.exceptionHandler.customExceptions.ParsingFormDataException;
+import com.kuchumov.springApp.exceptionHandler.customExceptions.ZipCreatingException;
+import com.kuchumov.springApp.repository.DataModelRepository;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.persistence.EntityNotFoundException;
+import javax.transaction.Transactional;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+@Service
+@Data
+@NoArgsConstructor
+public class DataService {
+    private DataModelRepository dataModelRepository;
+
+    @Autowired
+    public DataService(DataModelRepository dataModelRepository) {
+        this.dataModelRepository = dataModelRepository;
+    }
+
+    public EmptyResponseDTO saveFiles(NewDataURLDTO newDataURLDTO) {
+        List<DataModel> dataModelList = mapDataURLDTOToModel(newDataURLDTO);
+        dataModelRepository.saveAll(dataModelList);
+        return new EmptyResponseDTO(HttpStatus.OK, "Saved successfully");
+    }
+
+    protected List<DataModel> mapDataURLDTOToModel(NewDataURLDTO newDataURLDTO) {
```

Comment:
Сделай маппинг через Mapstruct

Reply:
```suggestion
    protected List<DataModel> mapDataUrlDtoToModel(NewDataURLDTO newDataURLDTO) {
```
надо везде сделать по подобию, легче читать